### ISSUE TITLE
Using `step` in slicing gives unclear error

### ISSUE THREAD

**Issue by piyueh** [nv-legate/legate.numpy#35]
## Problem

Something like `a[::3]` does not work and gives `TypeError: '<' not supported between instances of 'NoneType' and 'int'`, which is unclear about what happened.

## To reproduce

1. `test.py`
   ```python
   from legate import numpy
   a = numpy.random.random(100)
   print(a[::3])
   ```
2. `legate --cpus 1 ./test.py -lg:numpy:test`

## Output
```
Traceback (most recent call last):
  File "<prefix>/lib/python3.8/site-packages/legion_top.py", line 408, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "<prefix>/lib/python3.8/site-packages/legion_top.py", line 200, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "./test.py", line 3, in <module>
    print(a[::3])
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/array.py", line 382, in __getitem__
    shape=None, thunk=self._thunk.get_item(key, stacklevel=2)
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/deferred.py", line 428, in get_item
    view, dim_map = self._get_view(key)
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/thunk.py", line 378, in _get_view
    view = (self._standardize_slice_key(key, 0),)
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/thunk.py", line 325, in _standardize_slice_key
    or (key.stop < 0 and -key.step > diff)
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

## Note
Not urgent because it does not fail silently and still gives some error after all.

**Comment by manopapad** [comment]
The type error is fixed as of 72b9afe. Now Legate reports the actual reason slicing with step > 1 isn't working (`NotImplementedError: implement partition by phase`).

**Comment by piyueh** [comment]
I confirm the master branch now reports `NotImplementedError`.
